services/motion/builtin/builtin_test.go

```diff
@@ -698,811 +700,252 @@ func TestMoveOnMapTimeout(t *testing.T) {
-		defer ms.Close(ctx)
-		success, err := ms.MoveOnGlobe(
-			ctx,
-			fakeBase.Name(),
-			dst,
-			90,
-			injectedMovementSensor.Name(),
-			nil,
-			&motion.MotionConfiguration{LinearMPerSec: math.NaN()},
-			nil,
-		)
-		test.That(t, err, test.ShouldBeError, errors.New("LinearMPerSec may not be NaN"))
-		test.That(t, success, test.ShouldBeFalse)
-	})
+		srvc, ok := ms.(*builtIn).visionServices[cfg.ObstacleDetectors[0].VisionServiceName].(*inject.VisionService)
+		test.That(t, ok, test.ShouldBeTrue)
+		srvc.GetObjectPointCloudsFunc = tc.getPCfunc
 
-	t.Run("ensure success to a nearby geo point", func(t *testing.T) {
-		injectedMovementSensor, _, fakeBase, ms := createMoveOnGlobeEnvironment(ctx, t, gpsPoint, nil, 5)
-		defer ms.Close(ctx)
-		motionCfg := &motion.MotionConfiguration{PositionPollingFreqHz: 4, ObstaclePollingFreqHz: 1, PlanDeviationMM: epsilonMM}
 		req := motion.MoveOnGlobeReq{
-			ComponentName:      fakeBase.Name(),
+			ComponentName:      kb.Name(),
 			Destination:        dst,
 			MovementSensorName: injectedMovementSensor.Name(),
-			Obstacles:          []*spatialmath.GeoObstacle{},
-			MotionCfg:          motionCfg,
+			MotionCfg:          cfg,
 			Extra:              extra,
 		}
-		mr, err := ms.(*builtIn).newMoveOnGlobeRequest(ctx, req, nil, 0)
+		executionID, err := ms.MoveOnGlobe(ctx, req)
 		test.That(t, err, test.ShouldBeNil)
 
-		test.That(t, mr.planRequest.Goal.Pose().Point().X, test.ShouldAlmostEqual, expectedDst.X, epsilonMM)
-		test.That(t, mr.planRequest.Goal.Pose().Point().Y, test.ShouldAlmostEqual, expectedDst.Y, epsilonMM)
+		timeoutCtx, timeoutFn := context.WithTimeout(ctx, time.Minute*5)
+		defer timeoutFn()
+		err = pollUntilMOGSuccessOrError(timeoutCtx, ms, motion.PlanHistoryReq{
+			ComponentName: req.ComponentName,
+			ExecutionID:   executionID,
+			LastPlanOnly:  true,
+		})
 
-		planResp, err := mr.Plan(ctx)
-		test.That(t, err, test.ShouldBeNil)
-		test.That(t, len(planResp.Waypoints), test.ShouldBeGreaterThan, 2)
+		if tc.expectedSuccess {
+			test.That(t, err, test.ShouldBeNil)
+		} else {
+			test.That(t, err.Error(), test.ShouldEqual, tc.expectedErr)
+		}
+	}
 
-		success, err := ms.MoveOnGlobe(
-			ctx,
-			fakeBase.Name(),
-			dst,
-			0,
-			injectedMovementSensor.Name(),
-			[]*spatialmath.GeoObstacle{},
-			motionCfg,
-			extra,
-		)
-		test.That(t, err, test.ShouldBeNil)
-		test.That(t, success, test.ShouldBeTrue)
-	})
+	for _, tc := range testCases {
+		c := tc // needed to workaround loop variable not being captured by func literals
+		t.Run(c.name, func(t *testing.T) {
+			t.Parallel()
+			testFn(t, c)
+		})
+	}
+}
 
-	t.Run("go around an obstacle", func(t *testing.T) {
-		injectedMovementSensor, _, fakeBase, ms := createMoveOnGlobeEnvironment(ctx, t, gpsPoint, nil, 5)
-		defer ms.Close(ctx)
-		motionCfg := &motion.MotionConfiguration{PositionPollingFreqHz: 4, ObstaclePollingFreqHz: 1, PlanDeviationMM: epsilonMM}
+func TestCheckPlan(t *testing.T) {
```

Comment:
It's kind of interesting that this is a motionplan level function that we have unit tests around in that package, but we are also testing here.  It looks like maybe these are testing different things though, so maybe the name of this test is just throwing me off.  @nfranczak I think you wrote these, can you help me understand how these all fit together?

Reply:
deleted